keps/sig-cluster-lifecycle/generic/1755-communicating-a-local-registry/README.md

```diff
@@ -0,0 +1,545 @@
+-->
+
+N/A
+
+## Implementation History
+
+<!--
+Major milestones in the life cycle of a KEP should be tracked in this section.
+Major milestones might include
+- the `Summary` and `Motivation` sections being merged signaling SIG acceptance
+- the `Proposal` section being merged signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+## Drawbacks
+
+<!--
+Why should this KEP _not_ be implemented?
+-->
+
+This KEP is contingent on the maintainers of local clusters agreeing to it
+(specifically: Kind, K3d, Minikube, and Microk8s), since this is really about
+better documenting what they're already doing.
+
+This KEP is very narrow in scope and lightweight in implementation. It wouldn't
+make sense if there was a more ambitious local registry proposal. It also
+wouldn't make sense if there was a more ambitious proposal for cluster feature
+discovery.
+
+## Alternatives
+
+<!--
+What other approaches did you consider and why did you rule them out?  These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+
+At the sig-cluster-lifecycle meeting, we talked about many alternative
+proposals for where this data would be stored, including:
+
+- Using annotations on a Namespace
+- Using a plain ConfigMap
+- Using a Custom Resource
+
+With this proposal, I tried to strike a right balance between a format that
+followed existing conventions, and a format that didn't require too much
+Kubernetes API machinery to implement. Everyone who needs to implement this
+prefer that we choose a solution that does not require any Kubernetes code
+dependencies like apimachinery.
+
+We had a longer discussion about whether this should be "inert" data,
+or whether there should be mechanisms for keeping it up to date. In some cases,
+there is a one-to-one mapping between the cluster config and the RegistryStatus.
+For example, on Kind, the containerd config has the registry host in it.
+We also discussed whether there should be a solution for reflecting parts
+of the container configuration outside the cluster, and let tools outside the cluster
+read that to infer the existence of a local registry.
+
+But ultimately, I decided that the semantics of how these values align changes a
+lot, and that relying on it might be unwise.
+
+In the future, this config might apply to remote clusters. For example, you
+might have a remote managed development cluster with an in-cluster registry. Or
+each developer might have their own namespace, with an in-cluster registry
+per-namespace. I think the current proposal could potentially expand to include
+more registry topologies. But to avoid over-engineering, I haven't planned too
+deeply for what that might look like.
```

Comment:
ideally there should be no personalizations in the doc "I / We", but given this is not a scientific paper i leave it to you if you think it's worth the refactor.